persist/sqlite/consensus.go

```diff
@@ -324,6 +325,42 @@ func addTransactionFields(tx *txn, txns []types.Transaction, scDBIds map[types.S
 	return nil
 }
 
+func addHostAnnouncements(tx *txn, timestamp time.Time, hostAnnouncements []chain.HostAnnouncement, v2HostAnnouncements []explorer.V2HostAnnouncement) error {
+	var hosts []explorer.Host
+	for _, announcement := range hostAnnouncements {
+		hosts = append(hosts, explorer.Host{
+			PublicKey:  announcement.PublicKey,
+			NetAddress: announcement.NetAddress,
+
+			KnownSince:       timestamp,
+			LastAnnouncement: timestamp,
+		})
+	}
+	if len(hosts) > 0 {
+		if err := addHosts(tx, hosts); err != nil {
+			return fmt.Errorf("failed to insert host info: %w", err)
+		}
+	}
+
+	var v2Hosts []explorer.Host
+	for _, announcement := range v2HostAnnouncements {
+		v2Hosts = append(v2Hosts, explorer.Host{
+			PublicKey:      announcement.PublicKey,
+			V2NetAddresses: []chain.NetAddress(announcement.V2HostAnnouncement),
+
+			KnownSince:       timestamp,
+			LastAnnouncement: timestamp,
+		})
+	}
+	if len(v2Hosts) > 0 {
```

Comment:
(nit) you could combine `hosts` and `v2Hosts`  and only call `addHosts` once to save a few lines. Also it might be a bit nicer to have `addHosts` return early if it is called with 0 hosts rather than checking every time you call it.